text/0000-experimental_keywords.md

```diff
@@ -0,0 +1,74 @@
+
+It is probable that usage of an experimental keyword after the final syntax has been stabilized will fire a warning that you should move to the final syntax, but this can be decided on a case by case basis for each language feature.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+From time to time you may see things like `r#$name` in Rust code. These are "experimental keywords".
+
+An experimental keyword represents accessing the "raw operations" of the compiler and language. They have an intentionally funny syntax because experimental keywords are not intended to be used directly in the long term. Instead, they're a way to add features to the language before we've decided on the final syntax for that feature.
+
+Most experimental keywords are given a final syntax intended for general, long-term use, some time after the feature itself becomes available in the Stable compiler. This way the Rust community has an opportunity to try it out and see how it feels, how they use it in practice, and so on. Some experimental keywords aren't ever given a direct syntax of their own, instead they're simply there so that proc-macros can expand to code that uses the keyword.
+
+[And here we'd ideally we'd add at least one or two real examples to the guide once we have some examples of this in practice]
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+This RFC alters the language so that `r # $ token` is an accepted token sequence.
+* It names an experimental keyword named `token`.
+* The experimental keyword named _must_ be known to the compiler. If you attempt to name an experimental keyword that the compiler doesn't recognize it is an error.
+* The exact meaning and usage for each experimental keyword depends on the language feature it's associated with, and is not specified here.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* It can slightly hurt Rust's stability story to deliberately introduce new keywords that are intended to become deprecated within a relatively short period of time.
+  * Mitigation 1: Experimental keywords that do become Stable would continue to work even once the final syntax is decided upon and also stabilized. Stable code would not break.
+  * Mitigation 2: The rather unusual `r#$` prefix stands out as "clearly unusual"
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+* Alternative: We could select _some other_ sequence of invalid tokens to use as a prefix for experimental keywords.
+  * `r#$` is used because it is close enough to the "raw literal" and "raw string" syntaxes that people are very likely to understand what's going on even if they're not familiar with a specific experimental keyword when they first see it.
+
+# Prior art
+[prior-art]: #prior-art
```

Comment:
There has been some negative experiences with exposing things under non-final names, such as [vendor prefixes](https://developer.mozilla.org/en-US/docs/Glossary/Vendor_Prefix) in CSS or [`X-` headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers) in HTTP.

Reply:
Did you mean for this to be in the Rationale section or the Drawbacks section?

Reply:
Also, might as well mention here that you’re basically reinventing stropping.

Reply:
@Lokathor I put the comment here because I though of it by trying to think of additional prior art.  So maybe mention them here, with notes about how the choices already mentioned in the rest of the PR avoid the problems they ran into.